bin/ksh93_prereq.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# This script is used for feature detections that are required to build ksh93
+set -x
+set -e
+
+# This script is run from an unspecified directory so we have to determine directory paths
+# http://mesonbuild.com/Reference-manual.html#run_command
+script_path=`realpath "$0"`
+bin_dir=`dirname "$script_path"`
+base_dir=`dirname "$bin_dir"`
+
+PATH=$bin_dir:$PATH
+
+iffe_tests=( cmds  dynamic  externs  locale  poll  rlimits  setjmp  sigfeatures  time  ttys )
+
+iffe_tests_2=( nvapi shellapi )
+
+iffe_tests_3=( nfsd acct execargs pstat )
+
+pushd "$base_dir/src/cmd/ksh93/features"
+
+for iffe_test in ${iffe_tests[@]}; do
+    iffe -v -X ast -X std -c 'cc' run $iffe_test
+done
+
+for iffe_test in ${iffe_tests_2[@]}; do
+    iffe -v -X ast -X std -c 'cc' run $iffe_test
+    cp "$base_dir/src/cmd/ksh93/features/FEATURE/$iffe_test" "$base_dir/src/cmd/ksh93/features/$iffe_test.h"
+done
+
+iffe -v -c 'cc' run "$base_dir/src/cmd/ksh93/features/math.sh" "$base_dir/cmd/ksh93/data/math.tab"
```

Comment:
This generates an empty table with Meson. @krader1961  Any ideas about it ?

Reply:
I'll take a look but I'm strongly inclined to simply omit this feature test and move the *math.c* module into the main ksh source tree. Any platform we care about today supports all the math functions this module requires. If this module fails to compile and produce a usable `ksh` binary then the system has serious problems vis-a-vis providing the features ksh requires. The only reason I did it this way in the Nmake build system was to minimize the footprint of the change while still shaving a whole minute off the build time.

Reply:
My opinion is that *src/cmd/ksh93/bltins/meson.build* should include `'bltins/math.c'` and the relevant `iffe` test in *bin/ksh93_prereq.sh* should be removed. Then `cp src/cmd/ksh93/data/math.c src/cmd/ksh93/bltins/`. As I said in my previous comment any platform ksh currently builds on provides all the necessary math functions so the *math.c* module should "just work" and thus there isn't any reason to perform the `iffe` dance for those functions. And if that isn't true then all the people who have argued with me that `ksh` should support math functions like `scalb()` even though ksh provides no fallback if the platform does not provide the function need to explain why we should support those platforms.

Reply:
I have manually copied it under ```src/cmd/ksh93/bltins/``` and added it to meson build. Also I left a todo note on the original file for removal.